robottelo/hosts.py

```diff
@@ -135,13 +136,20 @@ def __init__(self, hostname, auth=None, **kwargs):
         elif isinstance(auth, str):
             # key file based authentication
             kwargs.update({'key_filename': auth})
+        self._satellite = kwargs.get('satellite')
         super().__init__(hostname=hostname, **kwargs)
 
+    @property
+    def satellite(self):
+        if not self._satellite:
+            self._satellite = Satellite()
+        return self._satellite
+
     @property
     def nailgun_host(self):
         """If this host is subscribed, provide access to its nailgun object"""
         if self.subscribed:
```

Comment:
self.subscribed doesn't necessarily indicate that its subscribed to a satellite instance, or that `self.satellite` has been set.  Taking my comment about whether `self.satellite` property should automatically inject the default satellite or return `None`, and maybe this function should check `self.subscribed and self.satellite is not None`.

As far as maintaining current assumptions, you're on track. I think there's a bit of a problem with the assumptions already in `master` branch.